### PR TITLE
Add a task&target to update project configurations

### PR DESCRIPTION
**Pull request by ericstj**
Previously there was just a target to flag errors for projects.

This wasn't all that useful so I added a target to fix them.

/cc @weshaggard @joperezr

### CODE REVIEW

File: build.proj [hunk 1]
```diff
@@ -57,16 +57,25 @@
     <MSBuild Projects="@(ExternalProject)"
              ContinueOnError="ErrorAndContinue" />
   </Target>
-  
- <UsingTask TaskName="ValidateVSConfigurations" AssemblyFile="$(CoreFxToolsTaskDir)CoreFx.Tools.dll"/>
- <Target Name="ValidateVSConfigurations" DependsOnTargets="BuildCoreFxTools" Condition="'$(CheckProjectConfiguration)'=='true'">
+
+  <UsingTask TaskName="ValidateVSConfigurations" AssemblyFile="$(CoreFxToolsTaskDir)CoreFx.Tools.dll"/>
+  <Target Name="ValidateVSConfigurations" DependsOnTargets="BuildCoreFxTools">
```
Comment (weshaggard) [inline build.proj]: Probably isn't even worth keeping the validation. I would say we just always fix them as opposed to validate them.

Comment (ericstj) [inline build.proj]: Yeah, that was my thought as well.

Comment (ericstj) [inline build.proj]: Though I did keep it in case we wanted CI or something to do this.

Comment (weshaggard) [inline build.proj]: I'm fine keeping it for now we can ask @karajas to clean it up. It doesn't hurt anything just not necessary.

**Comment by ericstj** [comment]
I just removed it.

File: build.proj [hunk 2]
```diff
@@ -57,16 +57,16 @@
...
+  <UsingTask TaskName="ValidateVSConfigurations" AssemblyFile="$(CoreFxToolsTaskDir)CoreFx.Tools.dll"/>
+  <Target Name="UpdateVSConfigurations" DependsOnTargets="BuildCoreFxTools">
+    <Message Importance="High" Text="Updating configurations for projects ..." />
     <ItemGroup>
       <ProjectsToCheckHeaders Include="$(MSBuildThisFileDirectory)src/**/*.*proj" Exclude="@(ProjectExclusions)" />
     </ItemGroup>
-    <ValidateVSConfigurations ProjectsToValidate="@(ProjectsToCheckHeaders)" />
-    <Message Importance="High" Text="Validating configurations for projects ... Done." />
- </Target>
+    <ValidateVSConfigurations ProjectsToValidate="@(ProjectsToCheckHeaders)" UpdateProjects="true" />
```
Comment (weshaggard) [inline build.proj]: nit: We still might want to change the name from Validate to Update.

**Pull request merged**